ledger/complete/ledger.go

```diff
@@ -467,6 +472,12 @@ func (l *Ledger) ExportCheckpointAt(
 
 	l.logger.Info().Msgf("start running post-checkpoint reporters")
 
+	if noMigration {
+		// when there is no mgiration, we generate the payloads now before
+		// running the postCheckpointReporters
+		payloads = newTrie.AllPayloads()
```

Comment:
if there is no migration, we delay the reading of the payloads, which is time-consuming, until the checkpoint has been generated